lib/package-integrity/handler/integrity.ts

```diff
@@ -0,0 +1,287 @@
+import { execSync } from 'child_process';
+import * as fs from 'fs';
+import * as os from 'os';
+import * as path from 'path';
+import { Repository } from './repository';
+
+/**
+ * Published package.
+ */
+export interface PublishedPackage {
+
+  /**
+   * Name of the package as stored in the package manager.
+   */
+  readonly name: string;
+
+  /**
+   * Version of the package as stored in the package manager.
+   */
+  readonly version: string;
+}
+
+/**
+ * Integrity class for validating a local artifact against its published counterpart.
+ *
+ * Implementations differ based on the package manager in question.
+ */
+export abstract class ArtifactIntegrity {
+
+  /**
+   * Download a package to the target directory.
+   *
+   * @param pkg The package to download.
+   * @param target The directory to download to.
+   */
+  protected abstract download(pkg: PublishedPackage, target: string): void;
+
+  /**
+   * Extract the artifact into the target directory.
+   *
+   * @param artifact Path to an artifact file.
+   * @param target The directory to extract do.
+   */
+  protected abstract extract(artifact: string, target: string): void;
+
+  /**
+   * Parse a local artifact file name into a structured package.
+   *
+   * @param artifactName Base name of the local artifact file.
+   * @returns The package this artifact correlates to.
+   */
+  protected abstract parse(artifactName: string): PublishedPackage;
+
+  /**
+   * The file extenstion of artifacts produced for this check. (e.g 'whl')
+   */
+  protected abstract get ext(): string;
+
+  /**
+   * Validate a local artifact against its published counterpart.
+   *
+   * @param localArtifactDir The directory of the local artifact. Must contain exactly one file with the appropriate extenstion.
+   */
+  public validate(localArtifactDir: string) {
```

Comment:
So this one actually is a bit different. The directory being passed here is the `dist/js`, `dist/python` directories of the local artifacts. I don't want to pass a file because the caller doesn't know exactly how the files in those directories are called, that's the responsibility of the individual integrity checkers.